src/org/dux/stracetool/Tracer.java

```diff
@@ -63,14 +143,164 @@ public void run() {
+            proc.waitFor();
+        } else if (os.startsWith("Windows")) {
+            // turn on Process Monitor
+            Process proc1 = rt.exec("cmd /c src\\org\\dux\\stracetool\\start_trace.bat");
+            proc1.waitFor();
+
+            // run actual command to trace
+            Process proc = rt.exec((String[]) args.toArray(new String[args.size()]));
+            Tracer.StreamGobbler outputGobbler = new Tracer.StreamGobbler(proc.getInputStream());
+            Tracer.logger.debug("waiting for build to terminate");
+            outputGobbler.start();
+            // TODO Replace with Java 9/10 Process.pid() once those become more mainstream
+            int myPid = getPid(proc);
+            proc.waitFor();
+
+            // turn off Process Monitor
+            Process proc2 = rt.exec("cmd /c src\\org\\dux\\stracetool\\end_trace.bat");
+            proc2.waitFor();
+
+            // need to manually filter after the trace if on Windows -- SLOW
+            try {
+                Set<Integer> parentPids = new HashSet<>();
+                if (traceSubprocesses) {
+                    parentPids = findParentPids(myPid, "strace.csv");
+                }
+                FileReader fr = new FileReader("strace.csv");
+                BufferedReader br = new BufferedReader(fr);
+                PrintStream out = new PrintStream(new File(fileName));
+                while (br.ready()) {
+                    String line = br.readLine();
+                    String[] parts = skipBadLine(line);
+                    if (parts == null) {
+                        continue;
+                    }
+                    if (parentPids.contains(Integer.parseInt(parts[6])) ||
+                        Integer.parseInt(parts[2]) == myPid) {
+                        if (filterCalls) {
+                            // NOTE: files, symbolic links, and hard links seem
+                            // to all be created or read through this procedure:
+                            // QueryDirectory, CreateFile, QueryBasicInformationFile,
+                            // CloseFile, CreateFile, QueryStandardInformationFile,
+                            // ReadFile, CloseFile
+
+                            // So tracing any of these calls would be sufficient
+                            String call = parts[3];
+                            if (!(call.equals("CreateFile") || call.equals("Process Create"))) {
+                                continue;
+                            }
+                            // TODO: what to do about readlink?
+                        }
+                        out.println(line);
+                    }
+                }
+                fr.close();
+                File f = new File("strace.csv");
+                f.delete();
+            } catch (IOException ioe) {
+                // won't happen; we just created the file we are opening
+                Tracer.logger.debug("You were wrong...");
```

Comment:
Probably we can put a better error message here, even if we think it won't ever happen

Reply:
Ok, fixed this.